db/initdb/schema.sql

```diff
@@ -0,0 +1,95 @@
+--
+-- PostgreSQL database dump
+--
+
+-- Dumped from database version 10.4 (Debian 10.4-1.pgdg90+1)
+-- Dumped by pg_dump version 10.3 (Ubuntu 10.3-1)
+
+SET statement_timeout = 0;
+SET lock_timeout = 0;
+SET idle_in_transaction_session_timeout = 0;
+SET client_encoding = 'UTF8';
+SET standard_conforming_strings = on;
+SELECT pg_catalog.set_config('search_path', '', false);
+SET check_function_bodies = false;
+SET client_min_messages = warning;
+SET row_security = off;
+
+--
+-- Name: plpgsql; Type: EXTENSION; Schema: -; Owner:
+--
+
+CREATE EXTENSION IF NOT EXISTS plpgsql WITH SCHEMA pg_catalog;
+
+
+--
+-- Name: EXTENSION plpgsql; Type: COMMENT; Schema: -; Owner:
+--
+
+COMMENT ON EXTENSION plpgsql IS 'PL/pgSQL procedural language';
+
+
+SET default_tablespace = '';
+
+SET default_with_oids = false;
+
+--
+-- Name: users; Type: TABLE; Schema: public; Owner: artesia
+--
+
+CREATE TABLE public.users (
+    id bigint NOT NULL,
+    email character varying(256) NOT NULL,
```

Comment:
There are 2 additional concepts that I think might be missing. In Pleroma, they are called `nickname` and `ap_id`. Mastodon refers to the nickname frequently (for example: `meff@lmbo.party`) in the API, and ActivityPub uses the `ap_id` to refer to actors (for example: `https://lmbo.party/users/meff`). While we could use the email to refer to what Pleroma calls the `nickname`, I don't think we should name the field `email`.

Alternatively we could define those IDs in a join table if we think we'll be referring to the IDs separately often enough than the users themselves.

Reply:
The "email" here refers to the normal e-mail address users would use to get stuff like account confirmation e-mails and such, not how others in the fediverse address this mastodon account.

I originally just put email and password in this table because was thinking that'd be the bare minimum for being able to create and login users (since Mastadon has users log in by e-mail and password).

It looks like pleroma has users log in as username + password, which seems better to me. We'd still need to store the user's actual e-mail though. Do you have strong opinions on whether users should log in via username + password, email + password, or {username or email} + password?

Reply:
I think the best would be {username or email} + password. But what would we do with things like the `nickname` or the `ap_id`? Should those just be tied into a join table elsewhere?

Reply:
I think we could stick those onto this table. Not sure yet though.